### PR TITLE
Sgratzl/downloadcharts

### LINKED ISSUE #364: Download for correlation and heatmap

### PR DESCRIPTION
**Pull request by sgratzl**
closes #364 

screenshot: 
image

download:
Heatmap (16)

screenshot:
image

download:
Correlation Network (8)

### CODE REVIEW

**Comment by jeffbaumes** [comment]
Not sure if it is from this branch or another recent one, but I'm getting the following problem now. I get an empty Clean Up Table view and empty charts. Is this a valid issue or is something I'm doing out of sync (db schema? docker?)

image

**Comment by sgratzl** [comment]
> Not sure if it is from this branch or another recent one, but I'm getting the following problem now. I get an empty Clean Up Table view and empty charts. Is this a valid issue or is something I'm doing out of sync (db schema? docker?)

this branch doesn't touch the files you get an error. Have you recently switched branches. Maybe the hot reloading is not perfect. At least I cannot reproduce the error locally

**Comment by jeffbaumes** [comment]
Yeah, when I restarted the client and server builds just now, the error went away.

File: web/src/components/vis/Heatmap.vue [hunk 1]
```diff
@@ -454,6 +455,102 @@ export default {
       this.rnode = null;
       this.cnode = null;
     },
+
+    generateImage() {
```
Comment (jeffbaumes) [inline web/src/components/vis/Heatmap.vue]: This code looks new (not moved from elsewhere). Does the same thing need to now be rendered in two different ways to support the interactive view and download? Will code need to change in two places to keep things looking the same?

Comment (sgratzl) [reply web/src/components/vis/Heatmap.vue]: the heatmap is not just a simple SVG like other plots but
 * two SVGs for the dendograms
 * a HTML canvas  for the matrix
 * two divs with for the labels

Thus, cannot be handled in the same way as other plots

**Pull request merged**